File: greedy/question/1092.py
from collections import deque
import sys
# import time
input = sys.stdin.readline
N = int(input())
crains = list(map(int, input().split()))
crains.sort(reverse=True)
M = int(input())
boxes = list(map(int, input().split()))
boxes.sort(reverse=True)
boxes = deque[boxes]
# start = time.time()

def ispossible(A, B):
    if A < B:
        return False
    return True

if not ispossible(crains[0], boxes[0]):
    print(-1)
    exit(0)
# print(crains, boxes)
times = 0
temp = []
while boxes or temp:
    temp.clear()
    times += 1
    i = 0
    while i < len(crains):
        if boxes:
            value = boxes.pop()
            if crains[i] < value:
                temp.append(value)
            else:
                i += 1
        else:
            break
    while temp:
        boxes.append(temp.pop())
print(times)
# print(time.time() - start)

File: greedy/1092.cpp
#include <iostream>
#include <vector>
#include <algorithm>
#include <functional>
using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<int> crains(n);
    for (int i = 0 ; i < n ; ++i)
    {
        cin >> crains[i];
    }

    int m;
    cin >> m;
    vector<int> boxes(m);
    for (int i = 0 ; i < m ; ++i)
    {
        cin >> boxes[i];
    }
    
    sort(crains.begin(), crains.end(), greater<>());
    sort(boxes.begin(), boxes.end(), greater<>());
    
    if (crains[0] < boxes[0])
    {
        cout << -1;
        exit(0);
    }
    int time = 0;
    while (true)
    {
        time++;
        bool flag = false;
        for (int i = 0 ; i < crains.size() ; ++i)
        {
            if (boxes.size())
            {
                for (int j = 0 ; j < boxes.size() ; ++j)
                {
                    if (crains[i] >= boxes[j])
                    {
                        boxes.erase(boxes.begin() + j);
                        break ;
                    }
                }
            }
            else
            {
                flag = true;
                break;
            }
        }
        if (!boxes.size() || flag)
            break;
        
    }
    cout << time;
}